jetstream/bigquery_client.py

```diff
@@ -134,3 +136,34 @@ def delete_experiment_tables(
 
         for existing_table in existing_tables:
             self.delete_table(f"{self.project}.{self.dataset}.{existing_table}")
+
+    def experiment_table_first_updated(self, slug: str) -> Optional[datetime]:
+        """Get the timestamp for when an experiment related table was updated last."""
+        table_prefix = bq_normalize_name(slug)
+
+        job = self.client.query(
+            rf"""
+            SELECT
+                table_name,
+                REGEXP_EXTRACT_ALL(
+                    option_value,
+                    '.*STRUCT\\(\"last_updated\", \"([^\"]+)\"\\).*'
+                ) AS last_updated
+            FROM
+            {self.dataset}.INFORMATION_SCHEMA.TABLE_OPTIONS
+            WHERE option_name = 'labels' AND table_name LIKE "enrollments_{table_prefix}%"
+            """
+        )
+        result = list(job.result())
+
+        table_first_updated = None
+
+        for row in result:
+            if not len(row.last_updated):
+                continue
+            updated_timestamp = UTC.localize(datetime.utcfromtimestamp(int(row.last_updated[0])))
```

Comment:
It seems to me that we could leave out the `pytz` parts of this update without any consequence, but maybe there's something important there that I'm missing. Dates/times are always a bit confusing...

Reply:
`utcfromtimestamp` doesn't actually set the `tzinfo`: https://docs.python.org/3/library/datetime.html#datetime.datetime.utcfromtimestamp
it basically returns a `datetime` based on the UTC time, but once you compare it with timestamps that have timezone information it will throw an error

Reply:
Is there an advantage to using `pytz` for timezones over, e.g., `datetime.fromtimestamp(timestamp, tz=timezone.utc)` like the example in the link? Is it just a remnant of existing code (and maybe and older recommended way of handling timezones) or does it still provide value for us?

Reply:
I just did it the same way it's done in other parts of jetstream. Might be worth updating though